internal/compiler/parser/element.rs

```diff
@@ -572,7 +579,22 @@ fn parse_transition_inner(p: &mut impl Parser) -> bool {
 fn parse_function(p: &mut impl Parser) {
     let mut p = p.start_node(SyntaxKind::Function);
     if p.peek().as_str() == "public" {
-        p.consume()
+        p.consume();
+        if p.peek().as_str() == "pure" {
```

Comment:
It might be worth adding a few tests for this in the unit test section of this function.